File: docs/algorithms/code/algorithms/intersection.ts
function intersection(nums1: number[], nums2: number[]): number[] {
  const set = new Set<number>();
  const res: number[] = [];

  for (let num of nums1) {
    set.add(num);
  }

  for (let num of nums2) {
    if (set.has(num) && !res.includes(num)) {
      res.push(num);
    }
  }

  return res;
};

File: docs/algorithms/code/algorithms/intersection.go
// 思路：使用哈希表进行记录，然后进行对比
// 复杂度分析：时间复杂度为 O(n), 空间复杂度为 O(1)， 因为数字是固定的
func intersection(nums1 []int, nums2 []int) []int {
	res := make([]int, 0)
	hash := make(map[int]bool)

	for _, v := range nums1 {
		hash[v] = true
	}

	for _, v := range nums2 {
		if hash[v] {
			res = append(res, v)
			hash[v] = false
		}
	}

	return res
}